app-backend/export/src/main/scala/com/azavea/rf/export/Export.scala

```diff
@@ -0,0 +1,147 @@
+import geotrellis.raster._
+import geotrellis.raster.io._
+import geotrellis.raster.histogram.Histogram
+import geotrellis.raster.io.geotiff.{GeoTiff, MultibandGeoTiff}
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.tiling._
+import geotrellis.spark.io.hadoop._
+import geotrellis.spark.io.file._
+import geotrellis.spark.io.s3._
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.fs.Path
+import org.apache.spark._
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import cats.implicits._
+
+import java.util.UUID
+
+object Export extends SparkJob with LazyLogging {
+  /** Get a LayerReader and an attribute store for the catalog located at the provided URI
+    *
+    * @param exportLayerDef export job's layer definition
+    */
+  def getRfLayerManagement(exportLayerDef: ExportLayerDefinition)(implicit @transient sc: SparkContext): (FilteringLayerReader[LayerId], AttributeStore) =
+    exportLayerDef.ingestLocation.getScheme match {
+      case "s3" | "s3a" | "s3n" =>
+        val (bucket, prefix) = S3.parse(exportLayerDef.ingestLocation)
+        val reader = S3LayerReader(bucket, prefix)
+        (reader, reader.attributeStore)
+      case "file" =>
+        val reader = FileLayerReader(exportLayerDef.ingestLocation.getPath)
+        (reader, reader.attributeStore)
+    }
+
+  def exportProject(params: CommandLine.Params)(exportDefinition: ExportDefinition)(implicit @transient sc: SparkContext) = {
+    val wrappedConfiguration = HadoopConfiguration(S3.setCredentials(sc.hadoopConfiguration))
+    val (input, output) = exportDefinition.input -> exportDefinition.output
+    val rdds = input.layers.map { ld =>
+      val (reader, attributeStore) = getRfLayerManagement(ld)
+      val layerId = LayerId(ld.layerId.toString, input.resolution)
+      val md = attributeStore.readMetadata[TileLayerMetadata[SpatialKey]](layerId)
+      val hist = ld.colorCorrections.map { _ => attributeStore.read[Array[Histogram[Double]]](LayerId(layerId.name, 0), "histogram") }
+      val crs = output.crs.getOrElse(md.crs)
+
+      val query = {
+        val q = reader.query[SpatialKey, MultibandTile, TileLayerMetadata[SpatialKey]](layerId)
+        input.mask.fold(q)(mp => q.where(Intersects(mp.reproject(LatLng, md.crs))))
+      } result
+
+      val bandsQuery = query.withContext {
+        _.mapValues { tile =>
+          val ctile = (ld.colorCorrections |@| hist) map { _.colorCorrect(tile, _) } getOrElse tile
+
+          output.render.flatMap(_.bands.map(_.toSeq)) match {
+            case Some(seq) if seq.nonEmpty => ctile.subsetBands(seq)
+            case _ => ctile
+          }
+        }
+      }
+
+      output.rasterSize match {
+        case Some(rs) if md.crs != crs || rs != md.tileCols || rs != md.tileRows =>
+          bandsQuery.reproject(ZoomedLayoutScheme(crs, rs))._2
+        case _ if md.crs != crs =>
+          bandsQuery.reproject(ZoomedLayoutScheme(crs))._2
```

Comment:
You still need to make sure to grab tile size from the source data, can't rely on it being default here. Unspecified == unchanged in this case, not default.